lib/toxiproxy_ex.ex

```diff
@@ -327,39 +311,29 @@ defmodule ToxiproxyEx do
       ...>  nil
       ...> end)
   """
-  @spec apply!(toxic_collection(), (() -> any())) :: :ok
-  def apply!(%ToxicCollection{toxics: toxics}, fun) do
-    dups =
-      Enum.group_by(toxics, fn t -> [t.name, t.proxy_name] end)
-      |> Enum.map(fn {_group, toxics} -> toxics end)
-      |> Enum.filter(fn toxics -> length(toxics) > 1 end)
-
-    if Enum.empty?(dups) do
-      # Note: We probably don't care about the updated toxies here but we still use them rather than the one passed into the function.
-      toxics =
-        Enum.map(toxics, fn toxic ->
-          case Toxic.create(toxic) do
-            {:ok, toxic} -> toxic
-            :error -> raise ServerError, message: "Could not create toxic '#{toxic.name}'"
-          end
-        end)
+  @spec apply!(toxic_collection(), (-> result)) :: result when result: var
+  def apply!(%ToxicCollection{toxics: toxics}, fun) when is_function(fun, 0) do
```

Comment:
The important change here is to use `try`/`after` to make sure we do destroy the toxic.